### DISCUSSION TITLE
Error message "The NVM file ID must be set before serializing" when trying to restore NVM on new controller

### DISCUSSION THREAD

**Discussion by Aegis940** [zwave-js/zwave-js#7199] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

Hello, I have an error when trying to restore the NVM on new zwave controller 

1. Backup NVM on current controller
2. Shutdown Zwave API
3. Switch to new controller 
4. Restart Zwave API
5. Restore NVM

Error is display at the end of "Convert NVM progress" process

### Software versions

ui: 9.20.0.bf3bdc3
driver: 13.4.0
server: 1.38.0

Managed by the Jeedom solution

### Z-Wave Controller (Stick/Dongle/...)

source controller:
_Z-Stick Aeotec Gen5 ZW090 (firmware Gen5+)
FW: v1.2
SDK: v6.81.6_

Target controller:
_Zooz 800 ZST39 LR
FW: v1.50
SDK: v7.22.1_ 

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[z-ui_current.log](https://github.com/user-attachments/files/17166243/z-ui_current.log)

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment]
Please share NVM backups of both controllers in their current state.

**Reply by Aegis940** [reply]
Hi, 

NVM_20240927140420.bin is for Aeotec Gen5 ZW090
NVM_20240927172506.bin is for Zooz 800 ZST39 LR

[
[NVM_20240927172506.bin.txt](https://github.com/user-attachments/files/17167668/NVM_20240927172506.bin.txt)
[NVM_20240927140420.bin.txt](https://github.com/user-attachments/files/17167669/NVM_20240927140420.bin.txt)
](url)

**Comment by cadwizzard** [comment]
I also have backups if needed of a (functional 66 node) 500 network and the same network restored and backed up from a 700 controller. 
I tried restoring both to the ZAC93 800 series controller with this same error. The 800 had previously on firmware 1.20 and several months ago allowed backups to be restored to it. The network was working but I swapped back to gen500 due to the transmit lockup issues back then on earlier 800 firmware.
I purchased a brand new ZAC93 800 series  controller with no network previously on it, and  this same error occurred when trying to restore a previous backup. 
I have the backups from the ZAC93 800 series with network and no network too if needed. 
They all convert to json ok with nvmedit, but converting previous generations to 800 format fails with the same error now.


[NVM800_ZAC93_Blank.json](https://github.com/user-attachments/files/17216910/NVM800_ZAC93_Blank.json)

**Comment by AlCalzone** [comment, marked as answer]
Converted to issue: https://github.com/zwave-js/node-zwave-js/issues/7219
